src/symtab2gb/symtab2gb_main.cpp

```diff
@@ -0,0 +1,113 @@
+#include <fstream>
+#include <goto-programs/goto_convert_functions.h>
+#include <goto-programs/goto_model.h>
+#include <goto-programs/link_goto_model.h>
+#include <goto-programs/write_goto_binary.h>
+#include <iostream>
+#include <json-symtab-language/json_symtab_language.h>
+#include <string>
+#include <util/exception_utils.h>
+#include <util/invariant.h>
+#include <util/optional.h>
+
+static inline bool failed(bool error_indicator)
+{
+  return error_indicator;
+}
+
+static void run_symtab2gb(
+  const std::vector<std::string> &symtab_filenames,
+  const std::string &gb_filename)
+{
+  // try opening all the files first to make sure we can
+  // even read/write what we want
+  auto out_file = std::ofstream{gb_filename};
+  if(!out_file.is_open())
+  {
+    throw system_exceptiont{"couldn't open output file `" + gb_filename + "'"};
+  }
+  std::vector<std::ifstream> symtab_files;
+  for(auto const &symtab_filename : symtab_filenames)
+  {
+    auto symtab_file = std::ifstream{symtab_filename};
+    if(!symtab_file.is_open())
+    {
+      throw system_exceptiont{"couldn't open input file `" + symtab_filename +
+                              "'"};
+    }
+    symtab_files.emplace_back(std::move(symtab_file));
+  }
+
+  auto message_handler = stream_message_handlert{std::cerr};
+
+  auto const symtab_language = new_json_symtab_language();
+  symtab_language->set_message_handler(message_handler);
+
+  auto linked_goto_model = goto_modelt{};
+
+  for(std::size_t ix = 0; ix < symtab_files.size(); ++ix)
+  {
+    auto const &symtab_filename = symtab_filenames[ix];
+    auto &symtab_file = symtab_files[ix];
+    if(failed(symtab_language->parse(symtab_file, symtab_filename)))
+    {
+      throw invalid_source_file_exceptiont{
+        "failed to parse symbol table from file `" + symtab_filename + "'"};
+    }
+    auto symtab = symbol_tablet{};
+    if(failed(symtab_language->typecheck(symtab, "<unused>")))
+    {
+      throw invalid_source_file_exceptiont{
+        "failed to typecheck symbol table from file `" + symtab_filename + "'"};
+    }
+    auto goto_model = goto_modelt{};
```

Comment:
I'd prefer `goto_modelt goto_model;`

Reply:
I prefer the `auto` form because with that I can make all declarations look the same, but I don't mind changing it if it bothers you.

Reply:
It does bother _me_, but I'm just one voice here. If others like this better then there's no need to fight about it. (I actually don't think that all declarations looking the same is desirable - quite often I don't even care much about the name, just about the type.)